### PR TITLE
RHINENG-19924: Refactor compliance connector to remove v1 API calls

### PR DESCRIPTION
**Pull request by marleystipich2**
## Summary by Sourcery

Remove legacy Compliance API v1 support and enforce v2 issue ID format in compliance connector

Enhancements:
- Require ssgVersion parameter and throw a BadRequest error for deprecated v1 identifiers
- Always construct and use the v2 URI for getRule and simplify result parsing to use data directly

Tests:
- Remove v1-specific unit tests and update them to validate v2 ID validation and caching behavior
- Add tests for missing or empty ssgVersion errors and successful v2 data retrieval using security guide and rule endpoints

### CODE REVIEW

File: src/connectors/compliance/impl.unit.js [hunk 1]
```diff
@@ -3,335 +3,167 @@
...
-        await impl.getRule('xccdf_org.ssgproject.content_rule_sshd_disable_root_login', 'xccdf_org.ssgproject.content_benchmark_RHEL-8', '0.0.0');
-        cache.get.callCount.should.equal(6);
-        cache.setex.callCount.should.equal(1);
+        
+        // Mock API responses
+        http.onFirstCall().resolves(createMockSecurityGuideResponse());
+        http.onSecondCall().resolves(createMockRuleResponse());
+
+        const result = await impl.getRule(MOCK_RULE_ID, MOCK_SSG_REF_ID, MOCK_SSG_VERSION);
+
```
[the remark this answers is not in the document]
Comment (marleystipich2) [reply src/connectors/compliance/impl.unit.js]: I fixed the tests so that they won't break if cache implementation changes

**Review by rexwhite — APPROVED** [review]
👍🏻
